### PR TITLE
Include Python 3.4 and 3.5 in .travis.yml (from felixonmars), with passing unit tests

### PR DESCRIPTION
**Pull request by fuhrysteve**

### CODE REVIEW

File: flask_login/__init__.py [hunk 1]
```diff
@@ -28,15 +24,41 @@
...
-        REFRESH_MESSAGE_CATEGORY, ID_ATTRIBUTE, AUTH_HEADER_NAME)
-assert (LoginManager, UserMixin, AnonymousUserMixin)
-assert (user_logged_in, user_logged_out, user_loaded_from_cookie,
-        user_loaded_from_header, user_loaded_from_request,
-        user_login_confirmed, user_unauthorized, user_needs_refresh,
-        user_accessed, session_protected)
-assert (current_user, login_url, make_secure_token, login_fresh, login_user,
-        logout_user, confirm_login, login_required, fresh_login_required,
-        set_login_view, encode_cookie, decode_cookie, make_next_param)
+__all__ = [
```
Comment (alanhamlett) [inline flask_login/__init__.py]: Much better than assert :+1:

Comment (fuhrysteve) [inline flask_login/__init__.py]: I couldn't get the tests to pass with the `assert` - not sure if due to a newer version of pyflakes, newer version of python, or some other reason.. at any rate, I get this in python 3.4 - not sure about earlier versions:

```python
In [1]: assert((False,))
<ipython-input-1-b82c5342027e>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
```

Comment (alanhamlett) [inline flask_login/__init__.py]: Actually, `__all__` only takes strings so passing literals as variables here won't work. You can try by adding `from flask.ext.login import *` and notice it fails with this exception:

`TypeError (Item in ``from list'' not a string)`

Comment (alanhamlett) [inline flask_login/__init__.py]: The only problem I can think of is `__all__` can only be used inside `__init__.py`. Having it in `_compat.py` doesn't actually do anything other than satisfy pyflakes.

Comment (alanhamlett) [inline flask_login/__init__.py]: Everything looks good to me, using `__all__` is supported by pyflakes and we should have used it in the first place.

**Pull request merged**